recipes/automake/all/conanfile.py

```diff
@@ -34,107 +32,91 @@ def configure(self):
+            self.tool_requires("msys2/cci.latest")
 
     def package_id(self):
         del self.info.settings.arch
         del self.info.settings.compiler
         del self.info.settings.build_type
 
+    def layout(self):
+        basic_layout(self, src_folder="source")
+        self.cpp.package.includedirs = []  # KB-H071: It is a tool that doesn't contain headers, removing the include directory.
+
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     @property
     def _datarootdir(self):
-        return os.path.join(self.package_folder, "res")
+        return path.join(self.package_folder, "bin", "share")
 
     @property
     def _automake_libdir(self):
-        return os.path.join(self._datarootdir, "automake-{}".format(self._version_major_minor))
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        conf_args = [
-            "--datarootdir={}".format(tools.unix_path(self._datarootdir)),
-            "--prefix={}".format(tools.unix_path(self.package_folder)),
-        ]
-        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
-        return self._autotools
-
-    def _patch_files(self):
-        for patch in self.conan_data["patches"][self.version]:
-            tools.patch(**patch)
-        if self.settings.os == "Windows":
-            # tracing using m4 on Windows returns Windows paths => use cygpath to convert to unix paths
-            tools.replace_in_file(os.path.join(self._source_subfolder, "bin", "aclocal.in"),
-                                               "          $map_traced_defs{$arg1} = $file;",
-                                               "          $file = `cygpath -u $file`;\n"
-                                               "          $file =~ s/^\\s+|\\s+$//g;\n"
-                                               "          $map_traced_defs{$arg1} = $file;")
+        version = Version(self.version)
+        return path.join(self._datarootdir, f"automake-{version.major}.{version.minor}")
 
-    def build(self):
-        self._patch_files()
-        autotools = self._configure_autotools()
-        autotools.make()
-
-    def package(self):
-        self.copy("COPYING*", src=self._source_subfolder, dst="licenses")
-        autotools = self._configure_autotools()
-        autotools.install()
-        tools.rmdir(os.path.join(self._datarootdir, "info"))
-        tools.rmdir(os.path.join(self._datarootdir, "man"))
-        tools.rmdir(os.path.join(self._datarootdir, "doc"))
+    @lru_cache(1)
+    def _autotools(self):
+        autotool = Autotools(self)
+        autotool.configure()
+        autotool.make()
+        return autotool
```

Comment:
please follow the same patterns than other AutotoolsToolchain recipes in CCI.

Reply:
I copied this pattern from Another recipe. Can you point toward an example so I can follow that pattern